client/v2/broadcast/comet/comet.go

```diff
@@ -0,0 +1,197 @@
+	BlockByHash(ctx context.Context, hash []byte) (*coretypes.ResultBlock, error)
+	BlockResults(ctx context.Context, height *int64) (*coretypes.ResultBlockResults, error)
+	BlockchainInfo(ctx context.Context, minHeight, maxHeight int64) (*coretypes.ResultBlockchainInfo, error)
+	Commit(ctx context.Context, height *int64) (*coretypes.ResultCommit, error)
+	Tx(ctx context.Context, hash []byte, prove bool) (*coretypes.ResultTx, error)
+	TxSearch(
+		ctx context.Context,
+		query string,
+		prove bool,
+		page, perPage *int,
+		orderBy string,
+	) (*coretypes.ResultTxSearch, error)
+	BlockSearch(
+		ctx context.Context,
+		query string,
+		page, perPage *int,
+		orderBy string,
+	) (*coretypes.ResultBlockSearch, error)
+}
+
+var _ broadcast.Broadcaster = &CometBFTBroadcaster{}
+
+// CometBFTBroadcaster implements the Broadcaster interface for CometBFT consensus engine.
+type CometBFTBroadcaster struct {
+	rpcClient CometRPC
+	mode      string
+	cdc       codec.JSONCodec
+}
+
+// NewCometBFTBroadcaster creates a new CometBFTBroadcaster.
+func NewCometBFTBroadcaster(rpcURL, mode string, cdc codec.JSONCodec) (*CometBFTBroadcaster, error) {
+	if cdc == nil {
+		return nil, errors.New("codec can't be nil")
+	}
+
+	if mode == "" {
+		mode = BroadcastSync
+	}
+
+	rpcClient, err := rpchttp.New(rpcURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create CometBft RPC client: %w", err)
+	}
+
+	return &CometBFTBroadcaster{
+		rpcClient: rpcClient,
+		mode:      mode,
+		cdc:       cdc,
+	}, nil
+}
+
+// Consensus returns the consensus engine name used by the broadcaster.
+// It always returns "comet" for CometBFTBroadcaster.
+func (c *CometBFTBroadcaster) Consensus() string {
+	return cometBFTConsensus
+}
+
+// Broadcast sends a transaction to the network and returns the result.
+// returns a byte slice containing the JSON-encoded result and an error if the broadcast failed.
+func (c *CometBFTBroadcaster) Broadcast(ctx context.Context, txBytes []byte) ([]byte, error) {
+	if c.cdc == nil {
+		return []byte{}, fmt.Errorf("JSON codec is not initialized")
+	}
+
+	var broadcastFunc func(ctx context.Context, tx cmttypes.Tx) (*coretypes.ResultBroadcastTx, error)
+	switch c.mode {
+	case BroadcastSync:
+		broadcastFunc = c.rpcClient.BroadcastTxSync
+	case BroadcastAsync:
+		broadcastFunc = c.rpcClient.BroadcastTxAsync
+	default:
+		return []byte{}, fmt.Errorf("unknown broadcast mode: %s", c.mode)
+	}
+
+	res, err := c.broadcast(ctx, txBytes, broadcastFunc)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	return c.cdc.MarshalJSON(res)
+}
+
+// broadcast sends a transaction to the CometBFT network using the provided function.
+func (c *CometBFTBroadcaster) broadcast(ctx context.Context, txBytes []byte,
+	fn func(ctx context.Context, tx cmttypes.Tx) (*coretypes.ResultBroadcastTx, error),
+) (*apiacbci.TxResponse, error) {
+	bResult, err := fn(ctx, txBytes)
+	if errRes := checkCometError(err, txBytes); err != nil {
```

Comment:
is there any chance, checkCometError switch check return the default value which is nil here ? Just want to make sure the error is properly return